api/clients/eigenda_client_e2e_test.go

```diff
@@ -18,29 +19,71 @@ func init() {
-		StatusQueryRetryInterval: 5 * time.Second,
-		CustomQuorumIDs:          []uint{},
-		SignerPrivateKeyHex:      "2d23e142a9e86a9175b9dfa213f20ea01f6c1731e09fa6edf895f70fe279cbb1",
-		// Waiting for finality adds 12 minutes to the test, and is not necessary
-		// because we already test for this correct behavior in the unit tests using a mock disperser
-		// which is much faster.
-		WaitForFinalization: false,
+
+	t.Run("PutBlobWaitForConfirmationDepth0AndGetBlob", func(t *testing.T) {
+		t.Parallel()
+		logger := log.NewLogger(log.NewTerminalHandler(os.Stdout, true))
+		client, err := NewEigenDAClient(logger, EigenDAClientConfig{
+			RPC:                      "disperser-holesky.eigenda.xyz:443",
+			StatusQueryTimeout:       25 * time.Minute,
+			StatusQueryRetryInterval: 5 * time.Second,
+			CustomQuorumIDs:          []uint{},
+			SignerPrivateKeyHex:      "2d23e142a9e86a9175b9dfa213f20ea01f6c1731e09fa6edf895f70fe279cbb1",
+			WaitForFinalization:      false,
+			WaitForConfirmationDepth: 0,
+			SvcManagerAddr:           "0xD4A7E1Bd8015057293f0D0A557088c286942e84b",
+			EthRpcUrl:                "https://1rpc.io/holesky",
+		})
+		data := "hello world!"
+		assert.NoError(t, err)
+		blobInfo, err := client.PutBlob(context.Background(), []byte(data))
+		assert.NoError(t, err)
+		batchHeaderHash := blobInfo.BlobVerificationProof.BatchMetadata.BatchHeaderHash
+		blobIndex := blobInfo.BlobVerificationProof.BlobIndex
+		blob, err := client.GetBlob(context.Background(), batchHeaderHash, blobIndex)
+		assert.NoError(t, err)
+		assert.Equal(t, data, string(blob))
+	})
+
+	t.Run("PutBlobWaitForConfirmationDepth3AndGetBlob", func(t *testing.T) {
+		t.Parallel()
+		confDepth := uint64(3)
+		logger := log.NewLogger(log.NewTerminalHandler(os.Stdout, true))
+		client, err := NewEigenDAClient(logger, EigenDAClientConfig{
+			RPC:                      "disperser-holesky.eigenda.xyz:443",
+			StatusQueryTimeout:       25 * time.Minute,
+			StatusQueryRetryInterval: 5 * time.Second,
+			CustomQuorumIDs:          []uint{},
+			SignerPrivateKeyHex:      "2d23e142a9e86a9175b9dfa213f20ea01f6c1731e09fa6edf895f70fe279cbb1",
+			WaitForFinalization:      false,
+			WaitForConfirmationDepth: confDepth,
+			SvcManagerAddr:           "0xD4A7E1Bd8015057293f0D0A557088c286942e84b",
+			EthRpcUrl:                "https://1rpc.io/holesky",
+		})
+		data := "hello world!"
+		assert.NoError(t, err)
+		blobInfo, err := client.PutBlob(context.Background(), []byte(data))
+		assert.NoError(t, err)
+		batchHeaderHash := blobInfo.BlobVerificationProof.BatchMetadata.BatchHeaderHash
+		blobIndex := blobInfo.BlobVerificationProof.BlobIndex
+		blob, err := client.GetBlob(context.Background(), batchHeaderHash, blobIndex)
+		assert.NoError(t, err)
+		assert.Equal(t, data, string(blob))
+
+		// assert confirmation depth
+		blockNumCur, err := client.ethClient.BlockNumber(context.Background())
+		assert.NoError(t, err)
+		blockNumAtDepth := new(big.Int).SetUint64(blockNumCur - confDepth)
```

Comment:
how do you check at least confDepth is waited?

Reply:
Extended the comment: [b29d79a](https://github.com/Layr-Labs/eigenda/pull/821/commits/b29d79a7c017c832f4ff9eef59c38c9168dafa97)
But basically we check that the batch was onchain at `blockNumCur - confDepth`, meaning the batch was confirmed at least confDepth blocks ago.